test/drivers/test_suites/syscall_exit_suite/init_module_x.cpp

```diff
@@ -0,0 +1,69 @@
+#include "../../event_class/event_class.h"
+
+#if defined(__NR_init_module)
+
+#if defined(__NR_close) && defined(__NR_open) && defined(__NR_read)
+TEST(SyscallExit, init_moduleX_failure)
+{
+	auto evt_test = get_syscall_event_test(__NR_init_module, EXIT_EVENT);
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	/* Open /dev/urandom for reading */
+	int fd = syscall(__NR_open, "/dev/urandom", O_RDONLY);
+	assert_syscall_state(SYSCALL_SUCCESS, "open", fd, NOT_EQUAL, -1);
+
+	/* Read data from /dev/urandom */
+	const unsigned data_len = DEFAULT_SNAPLEN / 2;
+	char mock_img[data_len];
+	ssize_t read_bytes = syscall(__NR_read, fd, (void *)mock_img, data_len);
+	assert_syscall_state(SYSCALL_SUCCESS, "read", read_bytes, NOT_EQUAL, 0);
+
+	char mock_buf[] = "AAABAACAADAAEAAFAAGAAHAAIAAJAAKAALAAMAA\0";
```

Comment:
I would avoid `udev` here, i would use the same strategy used for `mock_buf` so just a `char` buffer + a random `len` in this way we can avoid extra syscalls, for example `open` is not supported on ARM64